src/nvim/msgpack_rpc/channel.c

```diff
@@ -313,34 +312,35 @@ static void handle_request(Channel *channel, msgpack_object *request)
     return;
   }
 
-  // Retrieve the request handler
   MsgpackRpcRequestHandler handler;
-  Array args = ARRAY_DICT_INIT;
+
   msgpack_object *method = msgpack_rpc_method(request);
+  handler = msgpack_rpc_get_handler_for(method->via.bin.ptr,
+                                        method->via.bin.size);
+
+  // check method
+  if (handler.fn == msgpack_rpc_handle_missing_method) {
+    const char *m = method->via.bin.size > 0 ?
+                    method->via.bin.ptr : "<empty>";
+    api_set_error(&error, kErrorTypeException, "Invalid method: %s", m);
+  }
+  // check method arguments
+  Array args = ARRAY_DICT_INIT;
+  if (!msgpack_rpc_to_array(msgpack_rpc_args(request), &args)) {
+    api_set_error(&error, kErrorTypeException, "Invalid method arguments");
```

Comment:
Missing check for `ERROR_SET(&error)`?
If both errors are possible at the same time, the second error could overwrite the first one, resulting in a memory leak.

Reply:
error is initialized at the top and if it is set we enter a block that returns, so i was thinking it was already checked; but i'll insert a check after initializing args! 
Can i check for leaks with the make file?

Reply:
I have an ASAN target in my makefile:
 rm -rf build && make CC=\`which clang\` CMAKE_EXTRA_FLAGS='-DCMAKE_BUILD_TYPE=Debug -DCLANG_ASAN_UBSAN=ON' -j5

But I don't know if we have any tests for requests that have more than one error (invalid method name and invalid method arguments).